File: python/04_conversation_export.py
"""Export conversation history (member app_session key + chat_history plan).

    export TONIA_API_KEY=tonia_sk_…
    python 04_conversation_export.py
"""

from __future__ import annotations

import json
import os

from tonia import Tonia

with Tonia(api_key=os.environ["TONIA_API_KEY"]) as client:
    exported = client.conversations.export()
    print(json.dumps(exported, indent=2, ensure_ascii=False))

File: typescript/04-conversation-export.ts
/**
 * Export conversation history (needs a member app_session key + chat_history plan).
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 04-conversation-export.ts
 */
import { Tonia } from "@tonia/sdk";

const client = new Tonia({ apiKey: process.env.TONIA_API_KEY });
const exported = await client.conversations.export();
console.log(JSON.stringify(exported, null, 2));
